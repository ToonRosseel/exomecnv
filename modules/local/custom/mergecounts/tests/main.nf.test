nextflow_process {

    name "Test Process CUSTOM_MERGECOUNTS"
    script "../main.nf"
    process "CUSTOM_MERGECOUNTS"

    test("homo_sapiens - counts") {

        when {
            process {
                """
                input[0] = Channel.of([
                    [ id: 'Sample1', chromosome:"test", sam:["Sample1", "Sample2", "Sample3", "Sample4", "Sample5"], fam:["Fam1", "Fam2", "Fam3", "Fam4", "Fam5"] ],
                    [
                        file(params.counts1, checkIfExists:true),
                        file(params.counts2, checkIfExists:true)
                    ]
                ])
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test("homo_sapiens - counts - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = Channel.of([
                    [ id: 'Sample1', chromosome:"test", sam:["Sample1", "Sample2", "Sample3", "Sample4", "Sample5"], fam:["Fam1", "Fam2", "Fam3", "Fam4", "Fam5"] ],
                    [
                        file(params.counts1, checkIfExists:true),
                        file(params.counts2, checkIfExists:true)
                    ]
                ])
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
