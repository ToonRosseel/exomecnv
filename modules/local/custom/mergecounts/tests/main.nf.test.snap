{
    "homo_sapiens - counts": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "Sample1",
                            "chromosome": "merged",
                            "samples": [
                                "Sample1",
                                "Sample2",
                                "Sample3",
                                "Sample4",
                                "Sample5"
                            ],
                            "families": [
                                "Fam1",
                                "Fam2",
                                "Fam3",
                                "Fam4",
                                "Fam5"
                            ]
                        },
                        "Sample1.merged.txt:md5,8958d6e5d097113f6af0111900d81332"
                    ]
                ],
                "1": [
                    "versions.yml:md5,8894fd2d212a06e7bd0a3d53a1d5e0ee"
                ],
                "merge": [
                    [
                        {
                            "id": "Sample1",
                            "chromosome": "merged",
                            "samples": [
                                "Sample1",
                                "Sample2",
                                "Sample3",
                                "Sample4",
                                "Sample5"
                            ],
                            "families": [
                                "Fam1",
                                "Fam2",
                                "Fam3",
                                "Fam4",
                                "Fam5"
                            ]
                        },
                        "Sample1.merged.txt:md5,8958d6e5d097113f6af0111900d81332"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,8894fd2d212a06e7bd0a3d53a1d5e0ee"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.0"
        },
        "timestamp": "2025-09-05T10:39:56.61406686"
    },
    "homo_sapiens - counts - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "Sample1",
                            "chromosome": "merged",
                            "samples": [
                                "Sample1",
                                "Sample2",
                                "Sample3",
                                "Sample4",
                                "Sample5"
                            ],
                            "families": [
                                "Fam1",
                                "Fam2",
                                "Fam3",
                                "Fam4",
                                "Fam5"
                            ]
                        },
                        "Sample1.merged.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,8894fd2d212a06e7bd0a3d53a1d5e0ee"
                ],
                "merge": [
                    [
                        {
                            "id": "Sample1",
                            "chromosome": "merged",
                            "samples": [
                                "Sample1",
                                "Sample2",
                                "Sample3",
                                "Sample4",
                                "Sample5"
                            ],
                            "families": [
                                "Fam1",
                                "Fam2",
                                "Fam3",
                                "Fam4",
                                "Fam5"
                            ]
                        },
                        "Sample1.merged.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,8894fd2d212a06e7bd0a3d53a1d5e0ee"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.0"
        },
        "timestamp": "2025-09-05T10:40:06.356812228"
    }
}